ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/YARN/configuration/yarn-env.xml

```diff
@@ -121,138 +139,232 @@
+        </entry>
+        <entry>
+          <value>false</value>
+          <label>Disabled</label>
+        </entry>
+      </entries>
+      <selection-cardinality>1</selection-cardinality>
+    </value-attributes>
+    <depends-on>
+      <property>
+        <type>container-executor</type>
+        <name>gpu_module_enabled</name>
+      </property>
+    </depends-on>
+    <on-ambari-upgrade add="false"/>
   </property>
+
+  <property>
+    <name>registry.dns.bind-port</name>
+    <display-name>RegistryDNS Bind Port</display-name>
+    <value>53</value>
+    <description>
+      The port number for the DNS listener. The default port is 53.
+    </description>
+    <on-ambari-upgrade add="false"/>
+  </property>
+
   <!-- yarn-env.sh -->
   <property>
     <name>content</name>
     <display-name>yarn-env template</display-name>
     <description>This is the jinja template for yarn-env.sh file</description>
     <value>
-export HADOOP_YARN_HOME={{hadoop_yarn_home}}
-USER="$(whoami)"
-export HADOOP_LOG_DIR={{yarn_log_dir_prefix}}/$USER
-export HADOOP_PID_DIR={{yarn_pid_dir_prefix}}/$USER
-export HADOOP_LIBEXEC_DIR={{hadoop_libexec_dir}}
-export JAVA_HOME={{java64_home}}
-export JAVA_LIBRARY_PATH="${JAVA_LIBRARY_PATH}:{{hadoop_java_io_tmpdir}}"
+      export HADOOP_YARN_HOME={{hadoop_yarn_home}}
+      USER="$(whoami)"
+      export HADOOP_LOG_DIR={{yarn_log_dir_prefix}}/$USER
+      export HADOOP_PID_DIR={{yarn_pid_dir_prefix}}/$USER
+      export HADOOP_SECURE_LOG_DIR={{yarn_log_dir_prefix}}/$USER
+      export HADOOP_SECURE_PID_DIR={{yarn_pid_dir_prefix}}/$USER
+      export HADOOP_LIBEXEC_DIR={{hadoop_libexec_dir}}
+      export JAVA_HOME={{java64_home}}
+      export JAVA_LIBRARY_PATH="${JAVA_LIBRARY_PATH}:{{hadoop_java_io_tmpdir}}"
+
+      # We need to add the EWMA and RFA appender for the yarn daemons only;
+      # however, HADOOP_ROOT_LOGGER is shared by the yarn client and the
+      # daemons. This is restrict the EWMA appender to daemons only.
+      export HADOOP_LOGLEVEL=${HADOOP_LOGLEVEL:-INFO}
+      export HADOOP_ROOT_LOGGER=${HADOOP_ROOT_LOGGER:-INFO,console}
+      export HADOOP_DAEMON_ROOT_LOGGER=${HADOOP_DAEMON_ROOT_LOGGER:-${HADOOP_LOGLEVEL},EWMA,RFA}
+
+      # User for YARN daemons
+      export HADOOP_YARN_USER=${HADOOP_YARN_USER:-yarn}
+      # resolve links - $0 may be a softlink
+      export HADOOP_CONF_DIR="${HADOOP_CONF_DIR:-$HADOOP_YARN_HOME/etc/hadoop}"
+      # some Java parameters
+      # export JAVA_HOME=/home/y/libexec/jdk1.6.0/
+      if [ "$JAVA_HOME" != "" ]; then
+      #echo "run java in $JAVA_HOME"
+      JAVA_HOME=$JAVA_HOME
+      fi
 
-# User for YARN daemons
-export HADOOP_YARN_USER=${HADOOP_YARN_USER:-yarn}
+      if [ "$JAVA_HOME" = "" ]; then
+      echo "Error: JAVA_HOME is not set."
+      exit 1
+      fi
 
-# resolve links - $0 may be a softlink
-export HADOOP_CONF_DIR="${HADOOP_CONF_DIR:-$HADOOP_YARN_HOME/etc/hadoop}"
+      JAVA=$JAVA_HOME/bin/java
+      JAVA_HEAP_MAX=-Xmx1000m
```

Comment:
Were we setting this before also?

Reply:
If XmX1000m has been default, its fine.

Reply:
if YARN_HEAPSIZE set, it will overwrite JAVA_HEAP_MAX=-Xmx1000m